### PR TITLE
adding dtype_backend=pyarrow in documentation

### PR DESCRIPTION
**Pull request by JumpingDino**
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Issue took from [here](https://github.com/noatamir/pyladies-aug23-sprint/issues/4)

### CODE REVIEW

**Comment by JumpingDino** [comment]
@phofl @noatamir this PR was done in the berlin meetup.
Tagging you as requested :)

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -492,6 +492,21 @@ Setting ``dtype_backend="numpy_nullable"`` will result in nullable dtypes for ev
 
 .. _io.categorical:
```
Comment (phofl) [inline doc/source/user_guide/io.rst:493]: I think you have to remove that one, it's duplicated now in line 508 as well

**Comment by JumpingDino** [comment]
just removed the line, is this okay?

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. If interested in continuing, please merge in the main branch, address any review comments and/or failing tests, and we can reopen.

**Pull request closed without merging**